eng/pipelines/eng-common-sync.yml

```diff
@@ -30,16 +30,38 @@ stages:
         displayName: Sync ${{ parameters.DirectoryToSync }} Directory
 
         steps:
+        - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+          - pwsh: |
+              $patchDir = "$(Build.ArtifactStagingDirectory)/patchfiles"
+              New-Item $patchDir -Type Directory
+              $patchfiles = git format-patch HEAD...origin/master -- "${{ parameters.DirectoryToSync }}"
+              Copy-Item -Path "$(Build.SourcesDirectory)/*.patch" -Destination $patchDir
```

Comment:
We should be a able to pass `--output-directory` directly to the format-patch command.

Reply:
Strangely that command did not work. I tried it both on local and on the remote server but it wont place the patch files in the specified directory. There might be an issue with git, or am just not doing it properly.

Reply:
I used `-o` successfully locally.

Reply:
Yes that only seems to work if the output directory is within the source. I was trying to out then in the Artifact Staging Directory.

Reply:
If we cannot get the output directory to work then we should do a move instead of a copy so that we leave the repo in a clean state.